app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx

```diff
@@ -24,31 +28,44 @@ export interface ButtonProps extends reactAriaComponents.ButtonProps {
 }
 
 const DEFAULT_CLASSES =
-  'flex cursor-pointer rounded-sm border border-transparent transition-opacity duration-200 ease-in-out'
+  'flex cursor-pointer rounded-full border border-transparent transition-opacity duration-200 ease-in-out'
 const FOCUS_CLASSES =
   'focus-visible:outline-offset-2 focus:outline-none focus-visible:outline focus-visible:outline-primary'
+const SUBMIT_CLASSES = 'bg-invite text-white opacity-80 hover:opacity-100'
+const CANCEL_CLASSES = 'bg-selected-frame opacity-80 hover:opacity-100'
+const DELETE_CLASSES = 'bg-delete text-white'
 const ICON_CLASSES = 'opacity-50 hover:opacity-100'
 const EXTRA_CLICK_ZONE_CLASSES = 'flex relative before:inset-[-12px] before:absolute before:z-10'
 const DISABLED_CLASSES = 'disabled:opacity-50 disabled:cursor-not-allowed'
+const SIZE_CLASSES = 'px-2 py-1'
 
 /**
  * A button allows a user to perform an action, with mouse, touch, and keyboard interactions.
  */
 export function Button(props: ButtonProps) {
-  const { className, children, icon, ...ariaButtonProps } = props
+  const { className, children, variant, icon, loading = false, ...ariaButtonProps } = props
 
-  const classes = clsx(DEFAULT_CLASSES, DISABLED_CLASSES, FOCUS_CLASSES, ICON_CLASSES)
+  const classes = clsx(DEFAULT_CLASSES, DISABLED_CLASSES, FOCUS_CLASSES, SIZE_CLASSES, {
+    [SUBMIT_CLASSES]: variant === 'submit',
+    [CANCEL_CLASSES]: variant === 'cancel',
+    [DELETE_CLASSES]: variant === 'delete',
+    [ICON_CLASSES]: variant === 'icon',
+  })
 
-  const childrenFactory = () => {
-    return icon != null ? (
-      <>
-        <div className={EXTRA_CLICK_ZONE_CLASSES}>
-          <SvgMask src={icon} />
-        </div>
-      </>
-    ) : (
-      children
-    )
+  const childrenFactory = (): React.ReactNode => {
```

Comment:
i'd prefer avoiding the function wrapper here. nested ternaries might not be ideal but i think it'll probably be less noisy than `(): type => {` + `}` + a `()` further down. and probably makes it clearer that this isn't a function that is supposed to be called multiple times.

Reply:
(the function wrapper here is fine of course - the difference was a lot more noticeable previously when there wasn't a nested ternary)